src/mouse_events.cpp

```diff
@@ -785,17 +787,23 @@ void mouse_handler::select_or_action(bool browse)
 	wb::future_map_if_active planned_unit_map;
 
 	if(game_lua_kernel* lk = pc_.gamestate().lua_kernel_.get()) {
-		lk->select_hex_callback(last_hex_);
+		lk->select_hex_callback(clicked_hex);
+		// The callback will typically modify clicked_hex, so make sure it's still valid
+		if(!pc_.get_map().on_board(clicked_hex)) return;
 	}
 
-	unit_map::iterator clicked_u = find_unit(last_hex_);
+	unit_map::iterator clicked_u = find_unit(clicked_hex);
 	unit_map::iterator selected_u = find_unit(selected_hex_);
 
 	if(clicked_u && (!selected_u || selected_u->side() != side_num_ ||
 	  (clicked_u->side() == side_num_ && clicked_u->id() != selected_u->id()))
 	) {
-		select_hex(last_hex_, false);
+		select_hex(clicked_hex, false);
 	} else {
+		// There's a small chance the last hex has changed since this function began.
+		// Make sure the move applies to the hex that was originally clicked!
+		// TODO: Will this cause problems with previous_hex_?
```

Comment:
This TODO still needs investigation.

Reply:
I'm not sure how to test it, but the only place it's used seems to be in `mouse_handler::current_unit_attacks_from`…

Reply:
So what does that mean? Could the user see that the red attack-from arrow is coming in from one direction, click OK to attack, and then have the unit attack from a different direction instead?

Reply:
I'm not sure, but… that seems like something that could be tested…

Reply:
It looks like this can indeed happen. Tested with the following callback:

```lua
wesnoth.game_events.on_mouse_action = function()
    wesnoth.interface.delay(500)
end
```

Start an attack, move the mouse randomly, and the unit moves to the wrong hex and the attack never actually starts. Even if the wrong hex happens to still be next to the attacked unit, it doesn't attack.

Reply:
That seems like it needs to be fixed as part of the "action can trigger on the wrong hex" bug.